both/collections/competition.js

```diff
@@ -46,13 +49,19 @@ Competitions.attachSchema({
     max: 23.5*60,
     defaultValue: 23.5*60,
     custom: function() {
-      // require the competition calendar end time to be after the start time
-      var calendarStartMinutes = this.field("calendarStartMinutes").value;
-      var calendarEndMinutes = this.value;
-      if(calendarEndMinutes <= calendarStartMinutes) {
+      var obj = validationObject(this, ['calendarStartMinutes', 'calendarEndMinutes']);
+
+      if(obj.calendarEndMinutes <= obj.calendarStartMinutes) {
         return "calendarEndIsNotBeforeStart";
       }
-      return null;
+
+      var error = null;
+      ScheduleEvents.find({competitionId: this.docId}).forEach(function(event) {
+        if(event.endMinutes() > obj.calendarEndMinutes) {
+          error = "laterExistingEvents";
```

Comment:
Yeah, _.any() is better. I rewrote it.

It's weird how `for` was the only loop I knew - and loved - for decades, and now I won't even consider it...